### DISCUSSION TITLE
Why is INSERT statement treated as a separate case ?

### DISCUSSION THREAD

**Discussion by junli1026** [taosdata/TDengine#6577] [Q&A]
Hi experts,
I went through the query code and find that, the INSERT statement is NOT part of sql.y and it is treated separately.
Is there any reason for this ?  

The reason why I ask is, INSERT statement can be like, "INSERT INTO table [column1, column2] QUERY".
The QUERY here could be complicated. 

-Jun

**Comment by yu285** [comment]
have you had answer now ?

**Reply by junli1026** [reply]
It has been a while I totally forget ....
